### ISSUE TITLE
"public_network_access_enabled" argument giving Unsupported argument error

### ISSUE THREAD

**Issue by lingaswamy123** [hashicorp/terraform-provider-azurerm#18456]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.9

### AzureRM Provider Version

3.0.0

### Affected Resource(s)/Data Source(s)

azurerm_key_vault

### Description

azurerm_key_vault is not supporting the argument "public_network_access_enabled" 

### Terraform Configuration Files

```hcl
resource "azurerm_key_vault" "keyvault" {
  name                          = var.key_vault_name
  location                      = var.key_vault_location
  resource_group_name           = var.key_vault_resource_group
  enabled_for_disk_encryption   = var.key_vault_enabled_disk_encryption
  tenant_id                     = var.key_vault_tenant_id
  soft_delete_retention_days    = var.key_vault_soft_delete_retention_days
  purge_protection_enabled      = var.key_vault_purge_protection_enabled
  public_network_access_enabled = var.key_vault_public_network_access_enabled

  sku_name = var.key_vault_sku_name

  access_policy {
    tenant_id = var.key_vault_tenant_id
    object_id = var.key_vault_tenant_object_id

    key_permissions = [
      "Get",
    ]

    secret_permissions = [
      "Get",
    ]

    storage_permissions = [
      "Get",
    ]
  }
}

resource "azurerm_private_endpoint" "vaultendpoint" {
  name                = var.key_vault_private_endpoint_name
  location            = var.key_vault_location
  resource_group_name = var.key_vault_resource_group
  subnet_id           = var.key_vault_subnet_id

  private_service_connection {
    name                           = var.key_vault_private_connection_name
    private_connection_resource_id = azurerm_key_vault.keyvault.id
    is_manual_connection           = false
    subresource_names = ["Vault"]
  }
}
```

### Debug Output/Panic Output

```shell
Error: Unsupported argument
│ 
│   on modules/key-vault/vault.tf line 10, in resource "azurerm_key_vault" "keyvault":
│   10:   public_network_access_enabled = var.key_vault_public_network_access_enabled
│ 
│ An argument named "public_network_access_enabled" is not expected here.
╵
##[error]Error: The process '/opt/hostedtoolcache/terraform/1.2.9/x64/terraform' failed with exit code 1
```

### Expected Behaviour

public_network_access_enabled should apply to false 

### Actual Behaviour

public_network_access_enabled is by default public

### Steps to Reproduce

terraform plan
terraform apply

**Comment by jrhunger** [comment]
That functionality was released in 3.21.0. If you are really using 3.0.0 you should update and try again.

**Comment by tombuildsstuff** [comment]
[As @jrhunger has mentioned above](https://github.com/hashicorp/terraform-provider-azurerm/issues/18456#issuecomment-1253767705) this functionality was released in v3.21.0 - so 3.9.0 won't have this - you'll need to upgrade the version of the Provider being used (v3.27.0 is the latest version, which we'd recommend using) to use this field.

However since this functionality is available and can be used by updating the version of the Provider being used, I'm going to close this feature request out.

Thanks!
